#! /bin/sh -e 
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Description: Works around a broken build system, namely the
## DP: Description: simple bootloader on power 3/4.
## DP: Misc: Pulled from the ubuntu tree.
## DP: Patch author: fabbione@ubuntu.com
## DP: Upstream status: FTBFS fix, guess it will be fixed upstream too

. $(dirname $0)/DPATCH

@DPATCH@
diff -urNad linux-source-2.6.10-2.6.10/arch/ppc/boot/simple/misc-prep.c /usr/src/dpatchtemp/dpep.cQRwcC/linux-source-2.6.10-2.6.10/arch/ppc/boot/simple/misc-prep.c
--- linux-source-2.6.10-2.6.10/arch/ppc/boot/simple/misc-prep.c	2004-12-24 22:33:51.000000000 +0100
+++ /usr/src/dpatchtemp/dpep.cQRwcC/linux-source-2.6.10-2.6.10/arch/ppc/boot/simple/misc-prep.c	2004-12-28 10:43:29.838010536 +0100
@@ -152,9 +152,11 @@
 		hold_residual->VitalProductData.Reserved5 = 0xdeadbeef;
 	}
 
+#if defined(CONFIG_6xx)
 	/* Now go and clear out the BATs and ensure that our MSR is
 	 * correct .*/
 	disable_6xx_mmu();
+#endif
 
 	/* Make r3 be a pointer to the residual data. */
 	return (unsigned long)hold_residual;
